inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,94 @@
+"""Mixin for file management."""
+import os
+import shutil
+import pathlib
+from absl import logging
+
+from inductiva.utils import files, misc, templates
+
+
+class FileManager:
+    """Class for file management."""
+    __root_dir = None  #pylint: disable=invalid-name
+
+    def __check_root_dir(self):
+        if self.__root_dir is None:
+            raise ValueError("Root directory not set.")
+
+    def set_root_dir(self, root_dir=None):
+        """Set the root directory for the manager."""
+
+        if root_dir is None:
+            logging.info("Root directory not set. "
+                         "Setting default root_dir to be `input_dir`.")
```

Comment:
`None` is because it has not yet been initialised. We can do it because we can. Users should not be able to uninitialise something that they are trying to initialise. They should know what they want to do when they call the method and not hand out the decision to the devs. If that is the case, they shouldn't be trying to set the root dir.

Reply:
Okay, that seems like a good pattern to keep in mind! Thanks for the explanation!